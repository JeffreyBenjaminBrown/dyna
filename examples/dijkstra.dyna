
% Dijkstra's algorithm for single-source shortest path
path(*start) min= 0 .
path(B) min= path(A) + edge(A,B).

start += "a".

edge("a","b") += 1 .
edge("b","d") += 1 .
edge("a","d") += 3 .
edge("a","c") += 1 .
edge("c","d") += 2 .

% Expected
%  path("a") = 0
%  path("b") = 1
%  path("c") = 1
%  path("d") = 2
