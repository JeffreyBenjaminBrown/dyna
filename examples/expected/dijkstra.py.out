
Solution
========
end => "MyHouse".
goal => 2410.
start => "FriendHouse".

edge/2
======
edge("BOS","JFK")         => 187.
edge("BOS","MIA")         => 1258.
edge("DFW","LAX")         => 1235.
edge("DFW","ORD")         => 802.
edge("FriendHouse","BOS") => 10.
edge("JFK","DFW")         => 1391.
edge("JFK","MIA")         => 1090.
edge("JFK","SFO")         => 2582.
edge("LAX","ORD")         => 1749.
edge("MIA","DFW")         => 1121.
edge("MIA","LAX")         => 2342.
edge("ORD","DFW")         => 802.
edge("ORD","MyHouse")     => 20.

path/1
======
path("BOS")         => 10.
path("DFW")         => 1588.
path("FriendHouse") => 0.
path("JFK")         => 197.
path("LAX")         => 2823.
path("MIA")         => 1268.
path("MyHouse")     => 2410.
path("ORD")         => 2390.
path("SFO")         => 2779.

